moss_test.go

```diff
@@ -1,51 +1,65 @@
 package gomoss
 
-import "testing"
+import (
+	"bytes"
+	"os"
+	"testing"
+)
 
 func TestMossInit(t *testing.T) {
-    mossID := uint32(12345678)
+	mossID := uint32(12345678)
 	request := NewRequest(mossID)
 
-    switch {
-    case request.ID != mossID:
-        t.Fail()
-    case request.IgnoreLimit != 10:
-        t.Fail()
-    case request.ExperimentalServer != 0:
-        t.Fail()
-    case request.MatchingNumber != 250:
-        t.Fail()
-    case request.DirectoryMode != 0:
-        t.Fail()
-    case request.Language != "c":
-        t.Fail()
-    }
+	switch {
+	case request.ID != mossID:
+		t.Fail()
+	case request.IgnoreLimit != 10:
+		t.Fail()
+	case request.ExperimentalServer != 0:
+		t.Fail()
+	case request.MatchingNumber != 250:
+		t.Fail()
+	case request.DirectoryMode != 0:
+		t.Fail()
+	case request.Language != "c":
+		t.Fail()
+	}
 }
 
 func TestAddCode(t *testing.T) {
-    request := NewRequest(uint32(12345678))
+	moss := NewRequest(uint32(12345678))
 
-    request.Code.Add("aaa.c", "aaa.c")
-    if !(request.Code.sourceFile[0] == "aaa.c") || !(request.Code.displayName[0] == "aaa.c") {
-        t.Fail()
-    }
+	aFile, err := os.Open("testdata/aaa.c")
+	if err != nil {
+		t.Fail()
```

Comment:
you should use `t.Errorf` everywhere, except in cases like this, where a precondition is not met and the test is meaningless if it tries to continue. Here you should use `t.Fatalf` instead, which terminates the test immediately.